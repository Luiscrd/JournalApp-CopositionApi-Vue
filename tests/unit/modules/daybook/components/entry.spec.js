import { shallowMount } from '@vue/test-utils'
import Entry from '@/modules/daybook/components/Entry.vue'

describe('Pruebas en el Fab component', () => {

    const mockRouter = { push: jest.fn() }

    const entry = {
        id: '-NG3eMdNEqhIeBgnDk10',
        date: 1667599260784,
        picture: 'https://res.cloudinary.com/dtkdrn7w2/image/upload/v1667599268/tqvw4gix5okwge6myknb.png',
        text: 'Hola mundo'
    }

    let wrapper

    beforeEach(() => {

        wrapper = shallowMount(Entry, { 
            props: { 
                entry 
            },
            global: {
                mocks: {
                    $router: mockRouter
                }
            }
        })

    })

    test('Debe renderizar el componente correctamente', () => {

        expect(wrapper.html()).toMatchSnapshot()

    });

    test('Debe redirecionar al hacer click en el container', () => {

        const divPrin = wrapper.find('.entry-container')

        expect(divPrin.exists()).toBeTruthy()

        divPrin.trigger('click')

        expect(mockRouter.push).toHaveBeenCalled()

    });

    test('Pruebas de las propiedades computadas', () => {

        expect(wrapper.vm.day).toBe(4)

        expect(wrapper.vm.moth).toBe('Noviembre')

        expect(wrapper.vm.yearDay).toBe('2022, Viernes')
       
    });

})