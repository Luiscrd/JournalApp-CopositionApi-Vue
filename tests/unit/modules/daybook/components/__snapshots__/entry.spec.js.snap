// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Fab component Debe renderizar el componente correctamente 1`] = `
<div class="entry-container pb-3 p-2 pointer">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">4</span><span class="mx-1 fs-5">Noviembre</span><span class="mx-2 fw-light">2022, Viernes</span></div>
  <div class="entry-description">Hola mundo</div>
</div>
`;
